toucan_connectors/common.py

```diff
@@ -307,3 +308,26 @@ def adapt_param_type(params):
     postgres to correctly interpret them as an array.
     """
     return {k: (tuple(v) if isinstance(v, list) else v) for (k, v) in params.items()}
+
+
+def pandas_read_sql(
+    query: str,
+    con,
+    params=None,
+    adapt_params: bool = False,
+    convert_to_qmark: bool = False,
+    convert_to_printf: bool = True,
+    render_user: bool = False,
+    **kwargs,
+) -> pd.DataFrame:
+    if convert_to_printf:
+        query = convert_to_printf_templating_style(query)
+    if render_user:
+        query = Template(query).render({'user': params.get('user', {})})
+    if convert_to_qmark:
+        query, params = convert_to_qmark_paramstyle(query, params)
+    if adapt_params:
+        params = adapt_param_type(params)
```

Comment:
I'm bit dubious of factorization where we introduce a set of conditional branches. I also find that the the function name doesn't convey the sense of this operations.
But this is not blocking for now, as it's fairly easy to go back and fourth between implementations for now, and I don't have an alternative to suggest ^^